### ISSUE TITLE
The ExitStreamByteCount increments the wrong bin

### ISSUE THREAD

**Issue by teor2345** [privcount/privcount#276]
The ExitStreamByteCount counter inverts its arguments, so it increments the '1' bin by the total number of bytes.

This counter used to be called Stream[s]BytesAll.

**Comment by teor2345** [comment]
This issue also affects all the Exit*StreamByteCount counters.

**Comment by teor2345** [comment]
(But the Inbound, Outbound, and Ratio are correct.)

**Comment by teor2345** [comment]
I fixed this and added some checks so it's harder to mess up an increment() call

**Comment by teor2345** [comment]
@robgjansen, was this meant to be a single-bin counter?

**Comment by robgjansen** [comment]
Yes, this counter was counting the total number of bytes sent by the exit; see table 7 and table 10 in [the privcount paper](http://www.robgjansen.com/publications/privcount-ccs2016.pdf).

It was *not* intended to collect how many times the exit sent `n` bytes, as it would if it were a bin-type counter.

Sorry for the confusion!

**Comment by teor2345** [comment]
> Yes, it was meant to be a single-bin counter. This counter was counting the total number of bytes sent by the exit; see table 7 and table 10 in the privcount paper.
> 
> It was not intended to collect how many times the exit sent n bytes, as it would if it were a bin-type counter.

@robgjansen, I am going to assume that this also applies to Exit\*StreamOutboundByteCount and Exit\*StreamInboundByteCount, please let me know if I am wrong.

**Comment by robgjansen** [comment]
> @robgjansen, I am going to assume that this also applies to Exit*StreamOutboundByteCount and Exit*StreamInboundByteCount, please let me know if I am wrong.

You assumption is incorrect; I believe the code was intended as it is written. That is, the following increment the '1' bin by the byte amount (i.e., `num_increments=totalbw`) and so could be called 'single-counters':

```
self.secure_counters.increment('ExitWebStreamByteCount', 1, totalbw)
self.secure_counters.increment('ExitInteractiveStreamByteCount', 1, totalbw)
self.secure_counters.increment('ExitP2PStreamByteCount', 1, totalbw)
self.secure_counters.increment('ExitOtherPortStreamByteCount', 1, totalbw)
```

And the following increment the bin corresponding to the readbw or writebw amount by 1 (i.e., `num_increments=1`) and could be called 'bin-counters':

```
self.secure_counters.increment('ExitWebStreamOutboundByteCount', writebw)
self.secure_counters.increment('ExitWebStreamInboundByteCount', readbw)
self.secure_counters.increment('ExitInteractiveStreamOutboundByteCount', writebw)
self.secure_counters.increment('ExitInteractiveStreamInboundByteCount', readbw)
self.secure_counters.increment('ExitP2PStreamOutboundByteCount', writebw)
self.secure_counters.increment('ExitP2PStreamInboundByteCount', readbw)
self.secure_counters.increment('ExitOtherPortStreamOutboundByteCount', writebw)
self.secure_counters.increment('ExitOtherPortStreamInboundByteCount', readbw)
```

Is that clear?

We could add new counters so that we have the ability to count these bytes as single counters and as bin counters; I don't have a strong desire for that as long as we keep the currently implemented functionality.

**Comment by teor2345** [comment]
Ok, I made this fix in commit 763ef52.
